### ISSUE TITLE
Is Charging reports "Off" despite hot water heating

### ISSUE THREAD

**Issue by funks81** [tomasmcguinness/homeassistant-mixergy#61]
I have a mixergy connected to a Daikin heat pump - set as the default heat source (and should be only!!).  The "Is Charging" entity shows a state of 'Off' - despite the heat pump having been heated overnight by the heat pump (and also a little by the immersion - but that's a separate issue!!).

I also noticed the 'HeatPump Heat' and 'Electric Heat' entities don't seem to have a history - the history graph has a y-axis of 0 to 1.0.  I set a notification to alert me when the heat pump was using Electric Heat - and the automation did trigger when this was state 'on'.

Many Thanks for your help

![Screenshot 2024-10-07 at 13 00 37](https://github.com/user-attachments/assets/6263dc42-5b8d-486a-9f33-d271889433c9)
![Screenshot 2024-10-07 at 13 00 42](https://github.com/user-attachments/assets/1778a855-82d2-4a6c-a96a-71e00111972e)

**Comment by tomasmcguinness** [comment]
Hey.

I've found that the API responds in different ways to different heat sources, so it's possible the mechanism for detecting the IsCharging doesn't work correctly. 

As you are on a Heat Pump, I believe that the volumetric heating feature is disabled. Is that correct?

Electric Heat is the tank's internal element, so when you're using the heat pump, that won't register any heat. Does that help?

**Comment by tomasmcguinness** [comment]
Yeah, you're right, there is something wrong. I think those sensors are the wrong type. 

They should be boolean sensors, that are either *on* of *off* depending on the target charge level and the active hear source.

I'll have a dig!
